### PR TITLE
[POAE7-2934] Merge cider plan/processor to single cider_static library

### PR DESCRIPTION
**Pull request by xieqi**
### What changes were proposed in this pull request?

merge all cider library to a single one

### Why are the changes needed?

Refactor CMakeLists.txt

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Use exist Uts

### Which label does this PR belong to?

coderefactor

### CODE REVIEW

File: cpp/cmake-modules/substrait.cmake [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  set(PROTO_OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/thirdparty/substrait")
+  set(SUBSTRAIT_INCLUDE_DIR ${PROTO_OUTPUT_DIR})
+  set(substrait_proto_directory ${SUBSTRAIT_ROOT}/substrait)
+  file(MAKE_DIRECTORY ${PROTO_OUTPUT_DIR})
+  file(GLOB PROTO_FILES ${substrait_proto_directory}/*.proto
+       ${substrait_proto_directory}/extensions/*.proto)
+  foreach(PROTO ${PROTO_FILES})
+    file(RELATIVE_PATH REL_PROTO ${substrait_proto_directory} ${PROTO})
+    string(REGEX REPLACE "\\.proto" "" PROTO_NAME ${REL_PROTO})
```
Comment (Ma-Jian1) [inline cpp/cmake-modules/substrait.cmake:40]: seems we can delete those sed fixes in Makefile.

File: Makefile [hunk 2]
```diff
@@ -57,6 +57,7 @@ build-common:
 	@sed -i "s/velox\/substrait\/proto\///g" ${CPP_SOURCE_DIR}/thirdparty/velox/velox/substrait/proto/substrait/parameterized_types.proto
 	@sed -i "s/velox\/substrait\/proto\///g" ${CPP_SOURCE_DIR}/thirdparty/velox/velox/substrait/proto/substrait/plan.proto
 	@sed -i "s/velox\/substrait\/proto\///g" ${CPP_SOURCE_DIR}/thirdparty/velox/velox/substrait/proto/substrait/type_expressions.proto
```
Comment (Ma-Jian1) [inline Makefile:59]: can we make these fixes in the substrait.cmake?

File: cpp/src/cider-velox/benchmark/CMakeLists.txt [hunk 3]
```diff
@@ -50,11 +50,9 @@ set(FOLLY_WITH_DEPENDENCIES ${FOLLY_LIBRARIES} ${Boost_LIBRARIES}
 set(FOLLY_BENCHMARK Folly::follybenchmark)
 set(FOLLY ${FOLLY_LIBRARIES})
 set(VELOX_BENCHMARKS_DEPENDENCIES
-    cider_processor
-    cider_plan_substrait
+    cider_static
     velox_plan_transformer
     cider_plan_transformer
```
Comment (Ma-Jian1) [inline cpp/src/cider-velox/benchmark/CMakeLists.txt:55]: can we just export cider_velox static library, remove cider_plan_transformer and others in cider-velox

File: cpp/src/cider-velox/test/CMakeLists.txt [hunk 4]
```diff
@@ -101,10 +101,12 @@ set(VELOX_PLUGIN_TEST_LIBS
     ${FMT}
     ${FILESYSTEM}
     ${LLVM_LIB}
-    cider
-    cider_processor
-    cider_plan_substrait
-    cider_plan_parser
+    cider_static
```
Comment (Ma-Jian1) [inline cpp/src/cider-velox/test/CMakeLists.txt:104]: can we just use only cider_static outside?

File: cpp/src/cider/exec/nextgen/CMakeLists.txt [hunk 5]
```diff
@@ -17,7 +17,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
-set(NEXTGEN_DEPS cider_util gflags)
+set(NEXTGEN_DEPS cider_util ${GFLAGS_LIBRARIES})
```
Comment (Ma-Jian1) [inline cpp/src/cider/exec/nextgen/CMakeLists.txt:20]: what's the difference?

File: cpp/cmake-modules/substrait.cmake (hunk 1 above)
Comment (xieqi) [reply cpp/cmake-modules/substrait.cmake:40]: can't, these sed fixes are for velox build

File: Makefile (hunk 2 above)
Comment (xieqi) [reply Makefile:59]: same as above, it is for velox build

File: cpp/src/cider-velox/benchmark/CMakeLists.txt (hunk 3 above)
Comment (xieqi) [reply cpp/src/cider-velox/benchmark/CMakeLists.txt:55]: yes, it's the final goal, will provide a single library libvelox_plugin for presto integration and libcider for other developer

File: cpp/src/cider/exec/nextgen/CMakeLists.txt (hunk 5 above)
Comment (xieqi) [reply cpp/src/cider/exec/nextgen/CMakeLists.txt:20]: no difference, but prefer to CMake style

**Review by Ma-Jian1 — APPROVED** [review]

File: cpp/src/cider-velox/test/CMakeLists.txt [hunk 6]
```diff
@@ -101,10 +101,12 @@ set(VELOX_PLUGIN_TEST_LIBS
...
+    nextgen
+    QueryEngine
```
Comment (chaojun-zhang) [inline cpp/src/cider-velox/test/CMakeLists.txt:106]: it would be better to rename all of our library with 'cider' prefix?

**Review by chaojun-zhang — APPROVED** [review]

**Pull request merged**